File: DSA/Ayush_Mishra_590014844/Day_13_Question_2.py
from collections import Counter

def can_form_palindrome(s):
    freq = Counter(s)
    odd_count = sum(1 for count in freq.values() if count % 2)
    return odd_count <= 1

# Example usage
print(can_form_palindrome("aab"))   # True
print(can_form_palindrome("abc"))   # False

File: DSA/Ayush_Mishra_590014844/Day_13_Question_2.c
#include <stdio.h>
#include <string.h>

int canFormPalindrome(const char *str) {
    int freq[256] = {0}, oddCount = 0;
    for (int i = 0; str[i]; i++) {
        freq[(unsigned char)str[i]]++;
    }

    for (int i = 0; i < 256; i++) {
        if (freq[i] % 2) oddCount++;
    }

    return oddCount <= 1;
}

int main() {
    const char *test = "aab";
    printf("%s\n", canFormPalindrome(test) ? "true" : "false");
    return 0;
}
